src/gluonts/dataset/common.py

```diff
@@ -300,13 +299,13 @@ def write_table_from_records(
         """
         file_path = Path(file_path)
         msg = f"Converting to arrow dataset: {file_path}"
-        l: Optional[int] = None
+        length: Optional[int] = None
         try:
-            l = len(it)  # type: ignore
+            length = len(it)  # type: ignore
         except:
-            l = None
+            length = None
```

Comment:
This looks like passing `length` does nothing.

Reply:
It's used two lines later in: `for rec in tqdm(it, total=length, desc=msg):`

Reply:
Ah, sorry, I thought length was passed as an argument, which it is not.

We should not do the blank except however.